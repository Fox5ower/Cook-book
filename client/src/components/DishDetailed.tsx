import React, { Component } from 'react';
import IDish from "../interfaces/IDish";


interface MyProps {
    dish: IDish,
    toggleDescription: Function,
    showing: boolean
}

interface MyState {

}



class DishDetailed2 extends Component<MyProps, MyState> {

    render() {
        return (
            <div className="wrapper" style={this.props.showing ? { margin: "0 0 0 0", visibility: "visible" } : {}}>
                <div className="recipe">
                    <span className="close" onClick={() => this.props.toggleDescription()}></span>
                    <div className="recipe__body">
                        <div className="engreediants">
                            <span className="body__name">
                                Engreediants
                                </span>
                            <ul>
                                {this.props.dish.engreediants.map((el, i) => {
                                    return (<li key={i}>
                                        <span className="counter">
                                            {i + 1}
                                        </span>
                                        <span className="engreediant">
                                            {el}
                                        </span>
                                    </li>
                                    )
                                })}
                            </ul>
                        </div>
                        <div className="method">
                            <span className="body__name">
                                Method
                                </span>
                            <div className="method__text">
                                {this.props.dish.method}
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}


export default DishDetailed2;
