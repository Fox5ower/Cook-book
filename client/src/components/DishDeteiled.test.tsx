import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { render, cleanup } from '@testing-library/react'
import DishDetailed from './DishDetailed'
import { IntlProvider } from 'react-intl'
import English from '../languages/en.json'

afterEach(cleanup)

const fakeDish = {
  _id: '1',
  name: 'Test Name',
  category: 'Test Category',
  method: 'Test Method',
  description: 'Test Description',
  engreediants: ['Test 1', 'Test 2', 'Test 3'],
  language: 'Test Lang',
  image: '/test-image',
}

const toggleDescription = (): void => {}

it('Should renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <IntlProvider locale="en" messages={English}>
      <DishDetailed
        dish={fakeDish}
        toggleDescription={toggleDescription}
        showing={false}
      />
    </IntlProvider>,
    div
  )
})

it('Should renders with given props', () => {
  const { getByText } = render(
    <IntlProvider locale="en" messages={English}>
      <DishDetailed
        dish={fakeDish}
        toggleDescription={toggleDescription}
        showing={false}
      />
    </IntlProvider>
  )
  expect(getByText('Test 1')).toBeInTheDocument()
  expect(getByText('Test 2')).toBeInTheDocument()
  expect(getByText('Test 3')).toBeInTheDocument()
  expect(getByText('Test Method')).toBeInTheDocument()
})

it('Should match the snapshot', () => {
  const tree = renderer
    .create(
      <IntlProvider locale="en" messages={English}>
        <DishDetailed
          dish={fakeDish}
          toggleDescription={toggleDescription}
          showing={false}
        />
      </IntlProvider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
