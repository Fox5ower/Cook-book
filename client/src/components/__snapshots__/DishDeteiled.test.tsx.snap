// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div
  className="wrapper"
  style={Object {}}
>
  <div
    className="recipe"
  >
    <span
      className="close"
      onClick={[Function]}
    />
    <div
      className="recipe__body"
    >
      <div
        className="engreediants"
      >
        <span
          className="body__name"
        >
          Engredients
        </span>
        <ul>
          <li>
            <span
              className="counter"
            >
              1
            </span>
            <span
              className="engreediant"
            >
              Test 1
            </span>
          </li>
          <li>
            <span
              className="counter"
            >
              2
            </span>
            <span
              className="engreediant"
            >
              Test 2
            </span>
          </li>
          <li>
            <span
              className="counter"
            >
              3
            </span>
            <span
              className="engreediant"
            >
              Test 3
            </span>
          </li>
        </ul>
      </div>
      <div
        className="method"
      >
        <span
          className="body__name"
        >
          Method
        </span>
        <div
          className="method__text"
        >
          Test Method
        </div>
      </div>
    </div>
  </div>
</div>
`;
